src/agent/src/image_rpc.rs

```diff
@@ -235,9 +144,7 @@ impl ImageService {
 
         let cid = Self::cid_from_request(req)?;
         let image = req.get_image();
-        // Can switch to use cid directly when we remove umoci
-        let v: Vec<&str> = image.rsplit('/').collect();
-        if !v[0].is_empty() && v[0].starts_with("pause:") {
+        if !cid.is_empty() && cid.starts_with("pause:") {
```

Comment:
Since we added the cid check when it was derived from the image name and `:`is replaced with `_` in image name , if we switch to use cid, it may can not detect `pause:`  Sorry for misleading comments 
`// Can switch to use cid directly when we remove umoci`

Reply:
Good spot, I'll switch to check for `pause` just in case.

Reply:
FWIW i think assuming "pause:" is part of pause image name may be wrong is some cases like okd or openshift

Reply:
This might be worth raising as a separate issue as it would introduce the vulnerability of the pause image being provided by the host in non-peer pods CoCo. It might also help indicate why you guys have seen the behaviour of the pause image being pulled, rather than using the built in version if the image name doesn't match this?

Reply:
Yes, it's related i guess, i wasn't able to follow the whole workflow. AFAIU normally pause container should be as any other container, in crio's configuration you can even set custom pause_command, I'll open an issue once I'll figure out what exactly is the issue:P

Reply:
Why is the `cid.is_empty()` test needed if `cid.starts_with` would rule that case out anyway?

Reply:
Good point - I've remove the `is_empty()` check. Thanks!